linked_list/main.py

```diff
@@ -62,34 +67,72 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+        self.length= 0 
+        if elements:
+            for elem in elements:
+                self.append(elem)
+                
     def __str__(self):
-        pass
+        return "start: {}, end: {}".format(self.start.elem,self.end.elem)
 
     def __len__(self):
-        pass
+        return self.length
 
     def __iter__(self):
-        pass
+        return iter(self.start, self.end)
 
     def __getitem__(self, index):
-        pass
+        target = self.start
+        if index == 0:
+            return target
+        else:
+            for i in range(index):
+                target = target.next
+        return target
 
     def __add__(self, other):
-        pass
+        for elem in other:
+           self.append(elem)
+        return self
+        
 
     def __iadd__(self, other):
-        pass
+        self.append(other)
+        return self
 
     def __eq__(self, other):
-        pass
-
+        return self.start == other.start and self.end == other.end and self.length ==other.length
+    
```

Comment:
I feel like I'm misunderstanding how your objects work, but is this only checked the first and last elements of your list, and the length?